src/NCronJob/Scheduler/JobQueueManager.cs

```diff
@@ -55,7 +55,7 @@ public void RemoveQueue(string queueName)
 
                 jobQueue.Clear();
                 jobQueue.CollectionChanged -= CallCollectionChanged;
-                semaphores.Clear();
+                semaphores.TryRemove(queueName, out _);
```

Comment:
Given that semaphores are "borrowed" per queue, it seems pretty weird to drop them all when removing a single specified queue.

Reply:
This bid is mainly from @falvarez1 who did an amazing job.

I know we had some edge cases around long-running jobs that will block certain code paths. **But** I think you are right here. There is no reason to remove all semaphores and allow some arbitrary behavior in other queues.

So - I would roll with your change.